### PR TITLE
Fix compilation warnings and use new LibHac APIs for executable loading

### PR DESCRIPTION
**Pull request by Thog**
As title say.

### CODE REVIEW

File: Ryujinx.HLE/HOS/Services/Sockets/Nsd/Manager/FqdnResolver.cs [hunk 1]
```diff
@@ -32,7 +32,9 @@ public ResultCode GetSettingName(ServiceCtx context, out string settingName)
                     return ResultCode.Success;
                 }
 
+#pragma warning disable CS0162
                 return ResultCode.NullOutputObject;
+#pragma warning restore CS0162detected
```
Comment (AcK77) [inline Ryujinx.HLE/HOS/Services/Sockets/Nsd/Manager/FqdnResolver.cs:37]:
```suggestion
#pragma warning restore CS0162
```

File: Ryujinx.HLE/Loaders/Executables/KipExecutable.cs [hunk 2]
```diff
@@ -1,35 +1,81 @@
...
+        public ulong ProgramId { get; }
+        public byte Priority { get; }
+        public int StackSize { get; }
+        public byte IdealCoreId { get; }
+        public int Version { get; }
+        public string Name { get; }
 
-        public KipExecutable(IStorage inStorage) : base(inStorage)
+
+
```
Comment (AcK77) [inline Ryujinx.HLE/Loaders/Executables/KipExecutable.cs:33]: Remove the 2 extra empty lines

File: Ryujinx.HLE/Loaders/Executables/KipExecutable.cs [hunk 3]
```diff
@@ -1,35 +1,79 @@
 using LibHac;
```
Comment (Thealexbarney) [inline Ryujinx.HLE/Loaders/Executables/KipExecutable.cs:1]: None of these usings are needed
using LibHac;
using LibHac.FsSystem.Save;
using System.IO;

File: Ryujinx.HLE/HOS/ProgramLoader.cs [hunk 4]
```diff
@@ -1,4 +1,5 @@
 using ARMeilleure.Translation.PTC;
+using LibHac.Loader;
```
Comment (Thealexbarney) [inline Ryujinx.HLE/HOS/ProgramLoader.cs:2]: The added usings aren't needed
using LibHac.Loader;
using System.Text;

**Pull request merged**